src/test/java/io/electrum/vas/api/TestResourceClasses.java

```diff
@@ -0,0 +1,213 @@
+            // ensure that the matching interface method was called
+            try {
+               targettedInterfaceMethod.invoke(Mockito.verify(spiedInterfaceImpl), interfaceMethodParameters);
+            } catch (NotAMockException name) {
+               throw new Exception(
+                     "The resource class does not meet condition 3 of this method (The "
+                           + getResourceImplementationMethodName
+                           + "() method must return a spied instance of the interfaceName interface i.e. one created by calling Mockito.spy(Object) on an instance of the interfaceName).",
+                     name);
+            }
+         }
+      }
+   }
+
+   @Test
+   public void testCorrectResourceOperationsIdentified() throws Exception {
+      ensureResourceOperationMethodBackwardsCompatibility(
+            CorrectResourceTestClass.class,
+            "correctResourceMethod",
+            ResourceTestInterfaceImpl.class,
+            "correctBaseMethod");
+   }
+
+   @Test(expectedExceptions = { InvocationTargetException.class })
+   public void testIncorrectResourceOperationsIdentified() throws Exception {
+      ensureResourceOperationMethodBackwardsCompatibility(
+            CorrectResourceTestClass.class,
+            "incorrectResourceMethod",
+            ResourceTestInterfaceImpl.class,
+            "incorrectBaseMethod");
+   }
+
+   @Test
+   public void testMissingDefaultConstructor() throws Exception {
+      try {
+         ensureResourceOperationMethodBackwardsCompatibility(
+               IncorrectResourceTestClassWithoutDefaultConstructor.class,
+               "resourceMethod",
+               ResourceTestInterfaceImpl.class,
+               "incorrectBaseMethod");
+         Assert.fail("This should have failed due to a missing default constructor");
+      } catch (Exception e) {
+         Assert.assertTrue(e.getCause() instanceof NoSuchMethodException);
+         Assert.assertEquals(
+               e.getMessage(),
+               "The resource class does not meet condition 1 of this method (There must be a default constructor).");
+      }
+   }
+
+   @Test
+   public void testMissingGetResourceImplementationMethod() throws Exception {
+      try {
+         ensureResourceOperationMethodBackwardsCompatibility(
+               IncorrectResourceTestClassWithoutGetResourceImplementationMethod.class,
+               "resourceMethod",
+               ResourceTestInterfaceImpl.class,
+               "incorrectBaseMethod");
+         Assert.fail("This should have failed due to a missing getResourceImplementation method");
+      } catch (Exception e) {
+         Assert.assertTrue(e.getCause() instanceof NoSuchMethodException);
+         Assert.assertEquals(
+               e.getMessage(),
+               "The resource class does not meet condition 2 of this method (There must be a method, getResourceImplementation(), which returns an instance of the interfaceName interface).");
```

Comment:
```suggestion
               "There must be a method, getResourceImplementation(), which returns an instance of the interfaceName interface");
```

Or change the error message thrown from ensureResourceOperationMethodBackwardsCompatibility